src/main/java/org/folio/repository/CustomFieldsRepositoryImpl.java

```diff
@@ -61,18 +74,17 @@
   }
 
   /**
-   * Fetches custom field ids from the database by given id using sql LIKE statement
-   * @param customFieldId  - id of custom field
+   * Fetches the maximum custom field reference id from the database by given custom field name using sql regexp
+   * @param customFieldName  - name of custom field
    * @param tenantId - tenant id
    */
   @Override
-  public Future<Integer> countById(String customFieldId, String tenantId) {
+  public Future<Integer> maxRefId(String customFieldName, String tenantId) {
     Future<ResultSet> future = Future.future();
-    final String query = String.format(COUNT_CUSTOM_FIELDS_BY_ID, getCFTableName(tenantId));
-    logger.debug("Getting custom field ids by given id: {}.", customFieldId);
-    JsonArray parameters = new JsonArray().add(customFieldId + "_%");
-    PostgresClient.getInstance(vertx, tenantId).select(query, parameters, future);
-    return future.map(this::mapCount);
+    final String query = String.format(SELECT_REF_IDS, customFieldName, getCFTableName(tenantId));
```

Comment:
I know that all dangerous characters are removed from customFieldName before calling this method, but it would still feel more safe if we passed customFieldName as a parameter. 
I mean something like this
```
  public static final String REF_ID_REGEX = "(%s_[1-9]{1,})";

  static final String SELECT_REF_IDS = "SELECT unnest(regexp_matches(" + JSONB_COLUMN + " ->> 'refId', ?)) as values FROM %s";
```
```
 public Future<Integer> maxRefId(String customFieldName, String tenantId) {
    Future<ResultSet> future = Future.future();
    final String query = String.format(SELECT_REF_IDS, getCFTableName(tenantId));
    String refIdRegex = String.format(REF_ID_REGEX, customFieldName);
    JsonArray parameters = new JsonArray().add(refIdRegex);
    logger.debug("Getting custom field ids by given name: {}.", customFieldName);
    PostgresClient.getInstance(vertx, tenantId).select(query, parameters, future);
    return future.map(this::mapMaxId);
  }
```

Reply:
agree with you, updated, thanks @andrii-paias